src/Throttle/Throttler/ElasticWindowThrottler.php

```diff
@@ -101,7 +104,9 @@ public function count()
         }
 
         try {
-            $this->counter = $this->cache->get($this->key);
+            /** @var CacheCount $item */
```

Comment:
I think it would be a nice improvement to check whether these items contain the expected type:
```
if (!$item instanceof CacheCount) {
    $this->counter = 0;
}
```
Theoretically the cache could contain garbage or unsafe data.

Reply:
Perhaps something for a seperate PR though

Reply:
Yeah, this can be done, but in case it is a hit, it should get the actual saved value (best case)